src/crsAdapter/MerlinAdapter.js

```diff
@@ -323,26 +331,107 @@ class MerlinAdapter {
 
         xmlService.KindOfService = CONFIG.crs.serviceTypes.hotel;
         xmlService.Service = service.destination;
         xmlService.Accommodation = [service.roomCode, service.mealCode].filter(Boolean).join(' ');
+        xmlService.StealBoarding = service.roomQuantity;
+        xmlService.Occupancy = service.roomOccupancy;
         xmlService.FromDate = dateFrom.isValid() ? dateFrom.format(CONFIG.crs.dateFormat) : service.dateFrom;
         xmlService.EndDate = dateTo.isValid() ? dateTo.format(CONFIG.crs.dateFormat) : service.dateTo;
+        xmlService.TravellerAllocation = '1' + ((service.roomOccupancy > 1) ? '-' + service.roomOccupancy : '');
+
+        let startLineNumber = parseInt(travellerAssociation.substr(0, 1) || 0, 10);
+        let endLineNumber = parseInt(travellerAssociation.substr(-1) || 0, 10);
+
+        if (!startLineNumber) return;
+
+        do {
+            let traveller = (((xml.TravellerBlock || {}).PersonBlock || {}).PersonRow || [])[startLineNumber - 1];
+
+            if (!traveller) continue;
+
+            traveller.Salutation = void 0;
+            traveller.Name = void 0;
+            traveller.Age = void 0;
+        } while (++startLineNumber <= endLineNumber);
+    }
+
+    /**
+     * @private
+     * @param service object
+     * @param xmlService object
+     * @param xml object
+     */
+    assignChildrenData(service, xmlService, xml) {
+        if (!service.children) {
+            return;
+        }
+
+        const getNextEmptyTravellerIndex = () => {
+            xml.TravellerBlock = xml.TravellerBlock || { PersonBlock: void 0 };
+            xml.TravellerBlock.PersonBlock = xml.TravellerBlock.PersonBlock || { PersonRow: void 0 };
+            xml.TravellerBlock.PersonBlock.PersonRow = xml.TravellerBlock.PersonBlock.PersonRow || [];
+
+            let personRows = xml.TravellerBlock.PersonBlock.PersonRow;
+            let travellerIndex = void 0;
+
+            personRows.some((traveller, index) =>{
+                if (!traveller.Salutation && !traveller.Name && !traveller.Age) {
+                    travellerIndex = index;
+
+                    return true;
+                }
+            });
+
+            if (travellerIndex !== void 0) return travellerIndex;
+
+            personRows.push({
+                [CONFIG.builderOptions.attrkey]: {
+                    travellerNo: personRows.length + 1,
+                },
+            });
+
+            return personRows.length - 1;
+        };
+
+        let travellerLineNumber = void 0;
+
+        service.children.forEach((child) => {
+            let travellerIndex = getNextEmptyTravellerIndex();
+            let traveller = xml.TravellerBlock.PersonBlock.PersonRow[travellerIndex];
+
+            travellerLineNumber = travellerIndex + 1;
+
+            traveller.Salutation = CONFIG.crs.salutations.kid;
+            traveller.Name = child.name;
+            traveller.Age = child.age;
+        });
+
+        if (!travellerLineNumber) return;
```

Comment:
Could this be made into a separate function? Conditional returns in the middle of a function scare me.

Reply:
i understand. i could create a local function for it?!?